release/models/flex-algo/openconfig-flexalgo.yang

```diff
@@ -0,0 +1,397 @@
+      }
+      enum TE_METRIC {
+        value 2;
+        description
+          "TE default metric as defined in [RFC5305] to be used
+           during the calculation.";
+      }
+    }
+    description
+      "Type of metric to be used during a Flex Algo path
+       computation";
+  }
+
+  typedef flex-algo-id {
+    type uint8 {
+      range "128..255";
+    }
+    description
+      "Flex-Algorithm identifies which algorithm should be used during
+       path computation";
+  }
+
+  typedef flex-algo-calc-type {
+    type uint8 {
+      range "0..127";
+    }
+    default "0";
+    description
+      "Calculation type value ranges from 0 to 127 both inclusive
+       from the IGP Algorithm Types registry defined under
+       Interior Gateway Protocol (IGP)
+       Parameters IANA registries.
+       If the required calculation type is Shortest Path First,
+       the value 0 SHOULD appear in this field.";
+  }
+
+  grouping algorithm-attributes {
+    description
+      "All of the possible attributes for an flexible algorithm
+       definition";
+    leaf name {
+      type string;
+      description
+        "Flexible Algorithm name";
+    }
+    leaf flex-algo-id {
+      type flex-algo-id;
+      mandatory true;
+      description
+        "Flex Algorithm Identifier";
+    }
+    leaf flex-algo-description {
+      type string;
+      description
+        "Description of this flex-algo";
+    }
+    leaf metric-type {
+      type flex-algo-metric-type;
+      default "IGP";
+      description
+        "Indicates which is the corresponding metric type for
+         this algorithm";
+    }
+    leaf calc-type {
+      type flex-algo-calc-type;
+      description
+        "Indicates the calculation type, if advertised";
+    }
+    leaf priority {
+      type uint8;
+      default "237";
+      description
+        "Indicates the priority for this algorithm";
+    }
+
+    leaf-list exclude {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/flex-algo-admin-groups/" +
+          "admin-group/admin-group-name";
+      }
+      description
+        "extended-admin-groups to exclude in path calculation.";
+    }
+    leaf-list include-all {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/flex-algo-admin-groups/" +
+          "admin-group/admin-group-name";
+      }
+      description
+        "extended-admin-groups, all of which must be included";
+    }
+    leaf-list include-any {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/flex-algo-admin-groups/" +
+          "admin-group/admin-group-name";
+      }
+      description
+        "extended-admin-groups, one of which must be included";
+    }
+    leaf-list srlg-names {
+      type leafref {
+        path "../../../../flex-algo-global-attributes/srlgs/srlg/name";
+      }
+      description
+        "list of references to named SRLGs to be excluded";
+    }
+    leaf flags-tlv {
```

Comment:
Again, flgs-TLV is something tight to a FAD, not the algo as such. What you want is the "flags", so I would rename it to algo-flags or so.

Reply:
Sure!